src/ansible/roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,110 @@
+
+- name: Download software zipfile
+  get_url:
+    url: "{{ keycloak_url }}"
+    dest: /tmp
+
+- name: Unzip software
+  unarchive:
+    remote_src: yes
+    src: "/tmp/{{ zip_file }}"
+    dest: /opt/keycloak
+    owner: keycloak
+    group: keycloak
+    extra_opts:
+      - --strip-components=1
+
+- name: Change ownership of files in /data/certs
+  file:
+    path: "/data/certs/{{ item }}"
+    owner: keycloak
+    group: keycloak
+  loop:
+    - master.keycloak.test.crt
+    - master.keycloak.test.key
+
+- name: Add CA certificate to keystore
+  shell: |
+    keytool -noprompt -import \
+      -keystore /data/certs/master.keycloak.test.keystore \
+      -file /data/certs/ca.crt \
+      -alias ca.crt \
+      -trustcacerts -storepass {{ service.keycloak.admin_password }}
+  args:
+    creates: /data/certs/master.keycloak.test.keystore
+
+- name: Add Keycloak certificate to keystore
+  shell: |
+    keytool -noprompt -import \
+      -keystore /data/certs/master.keycloak.test.keystore \
+      -file /data/certs/master.keycloak.test.crt \
+      -alias master.keycloak.test.crt \
+      -trustcacerts -storepass {{ service.keycloak.admin_password }}
+
+- name: Create keycloak service env file
+  copy:
+    content: |
+      KEYCLOAK_ADMIN=admin
+      KEYCLOAK_ADMIN_PASSWORD={{ service.keycloak.admin_password }}
+      #KC_LOG_LEVEL=debug
+      KC_HOSTNAME={{ inventory_hostname }}
+      KC_HTTPS_CERTIFICATE_FILE=/data/certs/master.keycloak.test.crt
+      KC_HTTPS_CERTIFICATE_KEY_FILE=/data/certs/master.keycloak.test.key
+      KC_HTTPS_TRUST_STORE_FILE=/data/certs/master.keycloak.test.keystore
+      KC_HTTPS_TRUST_STORE_PASSWORD={{ service.keycloak.admin_password }}
+      KC_HTTP_RELATIVE_PATH=/auth
+    dest: /etc/sysconfig/keycloak
+
+- name: Create systemd keycloak service file
+  copy:
+    content: |
+      [Unit]
+      Description=Keycloak Server
+      After=network.target
+
+      [Service]
+      Type=idle
+      EnvironmentFile=/etc/sysconfig/keycloak
+
+      User=keycloak
+      Group=keycloak
+      ExecStart=/opt/keycloak/bin/kc.sh start
+      TimeoutStartSec=600
+      TimeoutStopSec=600
+
+      [Install]
+      WantedBy=multi-user.target
+    dest: /etc/systemd/system/keycloak.service
+
+- name: Reload systemd to pickup changes added for keycloak
+  systemd:
+    daemon_reload: yes
+
+- name: Run build step for Keycloak
+  shell: |
+    su - keycloak -c '''
+    export KEYCLOAK_ADMIN=admin
+    export KEYCLOAK_ADMIN_PASSWORD={{ service.keycloak.admin_password }}
+    export KC_HOSTNAME=$(hostname):8443
+    export KC_HTTPS_CERTIFICATE_FILE=/data/certs/master.keycloak.test.crt
+    export KC_HTTPS_CERTIFICATE_KEY_FILE=/data/certs/master.keycloak.test.key
+    export KC_HTTPS_TRUST_STORE_FILE=/data/certs/master.keycloak.test.keystore
+    export KC_HTTPS_TRUST_STORE_PASSWORD={{ service.keycloak.admin_password }}
+    export KC_HTTP_RELATIVE_PATH=/auth
+    /opt/keycloak/bin/kc.sh build
```

Comment:
Doesn't it make sense to build keycloak before you create the systemd unit?

Reply:
Moved.